packages/jellyfish-transaction/src/script/dftx/dftx_token.ts

```diff
@@ -117,3 +119,54 @@ export class CTokenUpdateAny extends ComposableBuffer<TokenUpdateAny> {
     ]
   }
 }
+
+/**
+ * Known as "std::variant<CScript>" in cpp.
+ */
+interface VariantScript {
+  variant: number
+  context: Script
+}
+
+/**
+ * Known as "std::variant<CScript>" in cpp.
+ *
+ * Composable VariantScript, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+class CVariantScript extends ComposableBuffer<VariantScript> {
+  composers (tb: VariantScript): BufferComposer[] {
+    return [
+      ComposableBuffer.uInt32(() => tb.variant, v => tb.variant = v),
+      ComposableBuffer.single<Script>(() => tb.context, v => tb.context = v, v => new CScript(v))
+    ]
+  }
+}
+
+/**
+ * TokenBurn DeFi Transaction
+ */
+export interface TokenBurn {
+  amounts: TokenBalanceUInt32[]
+  from: Script
+  burnType: number
```

Comment:
if the `burnType` value [can only be 0](https://github.com/DeFiCh/ain/blob/master/src/masternodes/mn_checks.cpp#L1255-L1256), it may be better IMO to limit the type to the values allowed for burnType (similar to [here](https://github.com/JellyfishSDK/jellyfish/blob/main/packages/jellyfish-transaction/src/script/dftx/dftx_governance.ts#L158-#L171)) instead of allowing any `number` to prevent situations where invalid values are sent, or at least have some documentation here to indicate what are the acceptable values.

similarly for the `variant` property of `VariantScript` as well

Reply:
Updated! lemmeknow if I missed anything or something's incorrect! 🙏